### ISSUE TITLE
moab_to_catalog should instantiate object with known storage dir

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/preservation_core_catalog#167]
To avoid looking at storage roots unnecessarily, instead of 

```ruby
current_version = Stanford::StorageServices.current_version(druid)
```

do 
```ruby
moab = Moab::StorageObject.new(druid, object_dir)
moab_version = moab.current_version_id  # this should be cheaper as it instantiates fewer objects to get version number
```

**Comment by ndushay** [comment]
similarly for object_size -- see sul-dlss/moab-versioning/issues/56.   (If not done together, create a separate ticket for PC code getting object_size efficiently when we know the storage dir, blocked by sul-dlss/moab-versioning/issues/56.
